### DISCUSSION TITLE
Pagination totalPages prop is missing

### DISCUSSION THREAD

**Discussion by danielbettech** [icflorescu/mantine-datatable#133] [Q&A]
Hello!

When I make a request to the server, I get back the total number of pages, so I can make another request when the client click the number 2, 3, 4 etc. page. 
I don't request all the records the first request. So I need a totalPages prop to set the pagination numbers.
The mantine native Pagination component has a **total** prop achieve this behavior. I don't see this in this lib.

Thanks!

**Comment by icflorescu** [comment]
Please read the docs at https://icflorescu.github.io/mantine-datatable/examples/pagination

**Reply by danielbettech** [reply]
I can't find that prop in here. I need the **total number of pages**.

page → the current page number
onPageChange → a callback that is executed when the user changes the current page
totalRecords → the total number of records in the dataset
recordsPerPage → the number of records per page

Not in here

**Reply by icflorescu** [reply]
The DataTable needs to know the total number of records, which is why it doesn't have a `totalPages` property (which would be imprecise if you'd want to compute the number of records), but `totalRecords` and `recordsPerPage`.

Can't you use `totalRecords` and `recordsPerPage` to achieve the desired result?

**Reply by danielbettech** [reply]
I added recordsPerPage and that made the job. Thank you!

**Reply by danielbettech** [reply]
Actually I have a bug in my table. I receive 11 records from my api and the pagination shows 2 pages. When I click the button #2 there are no records.

My api endpoint is waiting me to pass a pageNumber so it can send me back the correct data.

This are the props I was passing to the native pagination component from mantine before using this lib. And was working fine.

```jsx
<Pagination
  total={fetchedProducts?.totalPages}
  page={currentPage}
  onChange={(value) => currentPageSet(value)}
/>
```

In this image you can look 11 / 11 records (all of them visible in the page 1). But is displaying 2 pages.

<img width="1264" alt="image" src="https://user-images.githubusercontent.com/23063420/204436891-89838588-b359-4e8b-926b-bdba34b3d836.png">
